doc/decisions/STEPS.md

```diff
@@ -0,0 +1,98 @@
+# Steps
+
+This document describes all steps a decision can run through.
+
+```mermaid
+flowchart LR
+    s((Start)) --> Drafts --> In_Discussion --> In_Progress --> Decided --> Partially_Implemented --> Implemented
+
+    %% Shortcuts:
+    s --> Decided
+    Decided --> Implemented
+```
+
+Additionally, decision that are not yet "Decided" can be become "Rejected" or "Delayed" at any point.
+
+> The first PR for a decision must always create the decision in the "Drafts" state.
+> If during the reviews of this PR it becomes clear that the decision is further along, can be moved to "In Discussion" or "In Progress" before the merge.
+
+## Drafts
+
+> This step is highly recommended and it is even required if the problem is not yet clear to all the core developers.
+
+The first step is to create a PR with:
+
+- **one** decision, where at least the "Problem" is filled out and "Decision", "Rationale" and "Implications" are **not** yet filled out.
+- a link from [README.md](../README.md) from the "Drafts" section to this decision.
+- optional backlinks from related decisions.
+
+This step is very important:
+
+- it is brainstorming for completely new ideas.
+- it clarifies the scope of the decision.
+- it clarifies relation to other problems.
+
+Decisions will have much smoother further steps if this step is done carefully without prejudice.
+It is especially important that one shouldn't have a fixed mind-set about a preferred solution from the beginning.
+
+> Everyone must agree that the problem exists so that a decision PR in "Drafts" step can be merged.
+> At least the problem must be clear to everyone involved before the decision can leave the "Drafts" step.
+> It must be so clear that everyone would be able to describe a test case that shows if a solution fixes the problem.
+
+## In Discussion
+
+> This step is recommended if it is not yet clear to the core developers which solution is the best.
+
+Here you must ensure:
+
+- problem, constraint and assumptions are well-explained and sound
+- consistency with other decisions
+- links from/to related decisions are created
+- there are several considered alternatives, each with rationale and implication
+- "Decision", "Rationale" and "Implications" are **not** yet filled out if there are people arguing for different options
+
+Here the decision should not only have one decision but should describe several solutions.
+For each solution a proposal, rationale and optionally implications should be given.
+
+## In Progress
+
+- You must include all further alternative proposals made in the "Considered Alternatives" section.
+- Now it is allowed to have the decision from the previous round in the "Decision" section.
+
+> For decisions "In Progress", unlike "In Discussion", also the solution space next to the problem is clear.
```

Comment:
This doesn't help me one bit. I still don't understand the when "In Discussion" becomes "In Progress", but not "Decided" yet.

To me this feels like a valid step in the theoretical process, but doesn't fit with using PRs. IMO it sounds like a decision that has a decision (in the "Decision" section), but is not merged yet (because reviews aren't done) would be "In Progress". But that is a useless distinction. That only means you need to move the decision file, after the reviews are done.

When would we merge a decision as "In Progress"?

If I look at the current decision in [`doc/decisions/2_in_progress`](https://github.com/ElektraInitiative/libelektra/tree/master/doc/decisions/2_in_progress), the picture doesn't become much clearer. Many of them are "Decided" AFAIAC, some are even "Partially Implemented". Only `simplify_api.md` is still not decided IMO, but I'd say that's "In Discussion" because it is not clear what the exact API changes will be.

Reply:
> This doesn't help me one bit.

I elaborated the explanation in 5a330e8e85f47200d8dd965579f696855df00eac

> I still don't understand the when "In Discussion" becomes "In Progress", but not "Decided" yet.

Should be answered above. I feel like a merge point before we finally decide is cleaner.

> That only means you need to move the decision file, after the reviews are done.

For that we need good solutions anyway.

> When would we merge a decision as "In Progress"?

Always. It is a must, as it cannot be directly decided. (Unless it was directly target as decided for decisions where no discussion is needed. E.g. a decision needs another decision, split up of decisions etc..)

> Many of them are "Decided" AFAIAC, some are even "Partially Implemented".

These are left-overs before we had the decision process. I started cleanup in  #4690. The cleanup won't be finished with  #4690, though.

Reply:
I think I get the "big picture" now see https://github.com/ElektraInitiative/libelektra/pull/4639#discussion_r1027069545. What I still don't quite understand is: Why we need both "In Discussion" and "In Progress"? Is it just a bookkeeping thing to know how close to a decision we are?

Maybe different names would be better (open to other suggestions):
- "Drafts" -> "Problem Draft" (because the problem is still a draft)
- "In Discussion" -> "Solution Draft" (problem is done, but considered alternatives, i.e. solutions, are still drafts)
- "In Progress" -> "In Review" (because theoretically everything is done, the review process just hasn't come to a decision yet)

If we want to keep the current names, IMO "In Discussion" and "In Progress" should be swapped. First it is "In Progress", i.e. we are working on possible solutions, _then_ it is "In Discussion", where we discuss only which solution is best.

Reply:
This is excellent input. I fully agree not only that the names are confusingly worded but trying to find better names also also something else became clear to me: Drafts should not have **any** requirements to be accepted. E.g. #4683 clearly is a well-written draft (clear problem, elaborate considered alternatives etc.). Just because of our definition of Drafts (that reviewers must agree on the problem), the PR is blocked. This hampers the possibility to collect proposals for problems that might occur in future. If we would already have had this draft in our drawer before thinking of COW, we would have probably implemented COW differently. But as COW is done now, the previous interesting decision now doesn't have a clear problem anymore. I agree that "Rejecting" such a decision is the wrong word. So I think it should be allowed to be merged as draft. If later another decision would benefit from having this decision, we can easily refer to it again.

I updated the decision process in 7691244f94ce63b266396546edfae56028bc3099
@kodebach Please review.